### PR TITLE
Store: Add additional selectors for getting product category info independent of pagination

### PR DESCRIPTION
**Pull request by justinshreve**
This PR is a follow-up to #20420, and adds some additional selectors.

These additional selectors are patterned off of some of the [core term/taxonomy](https://github.com/Automattic/wp-calypso/blob/5f15f3365213253724ca0d8798c6cb8160f00bc9/client/state/terms/selectors.js) selectors. They provide a way to get result and loading state data for all requests made against a specific query. Put another way, if multiple pages of results are loaded, these selectors give us easy access to the loaded product categories, without having to use selectors against each page of results manually. I'm using this to build out #20295.

To Test:

* Make sure all tests pass: `npm run test-client client/extensions/woocommerce/state/`
* You can also spot check promotions and product management per #20420, but this PR shouldn't affect anything since it just adds additional selectors.

### CODE REVIEW

File: client/extensions/woocommerce/state/sites/product-categories/reducer.js [hunk 1]
```diff
@@ -111,10 +111,27 @@ export function total( state = 0, action ) {
...
 }
 
+/**
+ * Tracks the total number of pages for the current query.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function totalPages( state = 0, action ) {
```
Comment (timmyc) [inline client/extensions/woocommerce/state/sites/product-categories/reducer.js]: Should the default state be an `{}` here?

File: client/extensions/woocommerce/state/sites/product-categories/selectors.js [hunk 2]
```diff
@@ -141,3 +141,103 @@ export function getProductCategory(
...
+/**
+ * Returns true if currently requesting product categories for a query, excluding all known
+ * queried pages, or false otherwise.
+ *
+ * @param {Object} state Whole Redux state tree
+ * @param {Object} [query] Query used to fetch product categories. If not provided, API defaults are used.
+ * @param {Number} [siteId] Site ID to check. If not provided, the Site ID selected in the UI will be used
+ * @return {Boolean}       Returns true if currently requesting product categories for a query, excluding all known queried pages.
+ */
+export function areProductCategoriesLoadingIgnoringPage( state, query, siteId ) {
```
Comment (timmyc) [inline client/extensions/woocommerce/state/sites/product-categories/selectors.js]: Did you want to have `siteId = getSelectedSiteId( state )` here?

File: client/extensions/woocommerce/state/sites/product-categories/test/selectors.js [hunk 3]
```diff
@@ -194,4 +224,63 @@ describe( 'selectors', () => {
...
+		test( 'should be 2, the set total, if the categories are loaded.', () => {
+			expect( getProductCategoriesLastPage( loadedState, {}, 345 ) ).to.eql( 2 );
+		} );
+
+		test( 'should be null (default) when categories are loaded only for a different site.', () => {
+			expect( getProductCategoriesLastPage( loadedState, {}, 456 ) ).to.eql( null );
+		} );
+	} );
+
+	describe( '#areProductCategoriesLoadingIgnoringPage', () => {
```
Comment (timmyc) [inline client/extensions/woocommerce/state/sites/product-categories/test/selectors.js]: Should any of the ignoring page tests include a `page` in the query passed in, just to verify it still works as expected?

Comment (justinshreve) [reply client/extensions/woocommerce/state/sites/product-categories/test/selectors.js]: Good call. Done!

**Comment by justinshreve** [comment]
Thanks @timmyc! I addressed your feedback.

File: client/extensions/woocommerce/state/sites/product-categories/selectors.js [hunk 4]
```diff
@@ -178,7 +178,11 @@ export const getProductCategoriesLastPage = (
  * @param {Number} [siteId] Site ID to check. If not provided, the Site ID selected in the UI will be used
  * @return {Boolean}       Returns true if currently requesting product categories for a query, excluding all known queried pages.
  */
-export function areProductCategoriesLoadingIgnoringPage( state, query, siteId ) {
+export function areProductCategoriesLoadingIgnoringPage(
+	state,
+	query,
```
Comment (timmyc) [inline client/extensions/woocommerce/state/sites/product-categories/selectors.js]: Mentioned this in the parent PR too, not sure if we want to have `query = {}` here as well.

**Pull request merged**